src/server/services/wms/qgswmsgetlegendgraphics.cpp

```diff
@@ -153,10 +153,24 @@ namespace QgsWms
       throw QgsBadRequestException( QgsServiceException::QGIS_InvalidParameterValue,
                                     parameters[QgsWmsParameter::BBOX] );
     }
+    // If we have a contextual legend (BBOX is set)
+    // make sure WIDTH and HEIGHT are > 0, default to 800x600
+    if ( ! parameters.bbox().isEmpty() )
+    {
+      if ( parameters.width().isEmpty() )
+      {
+        parameters.set( QgsWmsParameter::WIDTH, 800 );
```

Comment:
I think that to be perfectly explicit, we could use SRCWIDTH in this case.

Reply:
Yes, I thought about it, I will change and test this.